test/files/run/t5717.scala

```diff
@@ -20,7 +20,8 @@ object Test extends StoreReporterDirectTest {
     val List(i) = filteredInfos
     // for some reason, nio doesn't throw the same exception on windows and linux/mac
     val path = if(util.Properties.isWin)"\\a" else "/a"
-    val expected = s"error writing ${testOutput.path}/a/B.class: Can't create directory ${testOutput.path}${path}" +
+    import File.pathSeparator
+    val expected = s"error writing ${testOutput.path}${pathSeparator}a${pathSeparator}B.class: Can't create directory ${testOutput.path}${pathSeparator}/a" +
       "; there is an existing (non-directory) file in its path"
     assert(i.msg == expected, i.msg)
```

Comment:
Why we need assertEquals, or even better, `specty` power assert as the fundamental building block.

Reply:
Also thanks to you and Lukas for not complaining about this test from 2013. For some reason, committed on my birthday. I guess my gift to myself was to try something in Scala.